tuxemon/core/components/event/actions/combat.py

```diff
@@ -250,4 +279,23 @@ def random_encounter(self, game, action):
             game.state_dict["WORLD"].menu_blocking = True
             player1.moving = False
             player1.direction = {'down': False, 'left': False, 'right': False, 'up': False}
+    
+    
+    def check_battle_legal(self, player):
```

Comment:
This is a good addition. Any redundant code should be made into a function for reuse-ability.